components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5269,4 +5301,24 @@ private void setUserNameWithDomain(Map<String, String> userClaimValues, Map<Stri
                     .getDomainQualifiedUsername()));
         }
     }
+
+    private boolean isUserContains(org.wso2.carbon.user.core.common.User coreUser, String user) throws
+            org.wso2.carbon.user.core.UserStoreException {
+
+        if (isLoginIdentifiersEnabled() &&
+                StringUtils.isNotBlank(getPrimaryLoginIdentifierClaim())) {
+            String primaryLoginIdentifier = carbonUM.getUserClaimValue(coreUser.getUsername(),
+                    getPrimaryLoginIdentifierClaim(), null);
+            if (StringUtils.isBlank(primaryLoginIdentifier)) {
+                return user.equalsIgnoreCase(coreUser.getUsername().indexOf(UserCoreConstants.DOMAIN_SEPARATOR) > 0
+                        ? coreUser.getUsername().split(UserCoreConstants.DOMAIN_SEPARATOR)[1] : coreUser.getUsername());
+            } else {
+                return user.equalsIgnoreCase(primaryLoginIdentifier.indexOf(UserCoreConstants.DOMAIN_SEPARATOR) > 0
+                        ? primaryLoginIdentifier.split(UserCoreConstants.DOMAIN_SEPARATOR)[1] : primaryLoginIdentifier);
```

Comment:
can primaryLoginIdentifier.split(UserCoreConstants.DOMAIN_SEPARATOR)[1] cause array index out of bound exception? if so, handle it properly

Reply:
`primaryLoginIdentifier.split(UserCoreConstants.DOMAIN_SEPARATOR)[1]` will only be executed if the condition `primaryLoginIdentifier.indexOf(UserCoreConstants.DOMAIN_SEPARATOR) > 0` takes the value as true. Therefore array index out of bound exception will not occur.